src/04-modality-specific-files/05-task-events.md

```diff
@@ -102,20 +102,27 @@ onset duration  trial_type  identifier  database  response_time
 5.6 0.6 sad AF01ANSA  kdef  1.739
 ```
 
-This file should be accompanied by a data dictionary.  
+This file should be accompanied by a data dictionary.
 
 ```Text
 sub-control01/
     func/
         sub-control01_task-emoface_events.json
 ```
 
-The `trial_type` and `identifier` columns from the `*_events.tsv` files might be described in this
-dictionary as follows. 
-Note that all other columns SHOULD also be described but are omitted for the sake
-of example.
+Specific fields related to stimulus presentation details SHOULD also be present in this data dictionary:
 
-```JSON  
+| Column name                  | Description                                                                                                                                                                                                                      |
+|------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| StimulusPresentationOS       | RECOMMENDED. Operating system used to run the stimuli presentation software.                                                                                                                                                     |
+| StimulusPresentationSoftware | RECOMMENDED. Name of the software that was used to present the stimuli ideally with its [Research Resource Identifier](https://scicrunch.org/resources) Examples: 'Psychtoolbox, RRID:SCR_002881' or 'PsychoPy, RRID:SCR_006571' |
+| StimulusPresentationVersion  | RECOMMENDED. Version of the software that was used to present the stimuli.                                                                                                                                                       |
+| StimulusPresentationCode     | RECOMMENDED. URL to code of that was used to present the stimuli.                                                                                                                                                                |
```

Comment:
I would also allow DOIs (or even recommend), to push people towards "archiving" their presentation code (e.g., via zenodo).

But of course a URL can also be used to encode the DOI like https://doi.org/10.5281/zenodo.3361717 ... so no strong feeling from my side to absolutely HAVE TO HAVE url and doi both.

Reply:
Good point. Will add that in the description and also make use a zenodo doi in the example.

Reply:
Tried to "nudge" for using DOI rather than URLs